.github/workflows/reusable-revised-its.yml

```diff
@@ -72,6 +72,15 @@ jobs:
           path: ./**/target
           key: maven-${{ runner.os }}-${{ inputs.build_jdk }}-targets-${{ github.sha }}
 
+      - name: Clean up docker
```

Comment:
For the "new ITs", the `it.sh` script itself should clean up. There is a bug that will be fixed in another PR, but once that is fixed, we want the script to do the work. We need it to work for developers, so we want it to work for builds also.

Given that, if there are any containers to stop in this script block, then that indicates a bug in our `it.sh` script. Thus, it would be good to

* Count the number of running containers.
* If more than 0:
   * Echo the list of running containers.
   * Do the force shutdown steps

Another question is the set of files not owned by `$USER_NAME`. What might cause that to occur? There are several possible cases:

* The cached files grabbed earlier are owned by another user. (This was the case on an older proprietary Jenkins setup.)
* The Docker containers created files owned by another user.

Both cases are bugs. Having mixed ownership will cause all manner of problems if it results in an inability to delete files, overwrite files or create directories. Thus, rather than simply clean up, we should log such files so we can fix them. This comment applies only to the "new IT" case: those run by `it.sh`. The old test may do all manner of horrible things.

Reply:
Maybe it would be more simple to just force stop and prune any containers matching the ones we know we'll be creating with the new ITs/`it.sh`?

The files not owned by root is a direct result of running in containers, files created in them will often be created as `root`, and thus the runner user (ubuntu in this case) doesn't have permission to clean them, and as a result, the GHA runner can't clean up properly and artifacts get left around.  I threw it in as a good measure, since it doesn't really add any time or risk.

Reply:
There's some options for running containers as a user, but they're not really available to us in the hosted runners, as we don't have control over the base image.  We could install our own docker, running in the user env, but that'd cost both time and effort that simply cleaning up like this solves easily.